tests/units/Glpi/Cache/CacheStorageFactory.php

```diff
@@ -0,0 +1,272 @@
+            'config'   => [
+                'adapter' => 'auto',
+                'options' => [
+                   'namespace' => 'app_cache',
+                ],
+            ],
+            'expected_adapter' => [
+               'class'   => \Zend\Cache\Storage\Adapter\Apcu::class,
+               'options' => [
+                  'namespace' => 'app_cache',
+               ],
+               'plugins' => [
+               ],
+            ],
+         ],
+
+         // Case: dba adapter using default options
+         /* Cannot test without extension loaded
+         [
+            'config'   => [
+                'adapter' => 'dba',
+            ],
+            'expected_adapter' => [
+               'class'   => \Zend\Cache\Storage\Adapter\Dba::class,
+               'options' => [
+                  'pathname' => GLPI_CACHE_DIR . '/_default.data',
+               ],
+               'plugins' => [
+                  'serializer'
+               ],
+            ],
+         ],
+         */
+
+         // Case: filesystem adapter using default options
+         [
+            'config'   => [
+                'adapter' => 'filesystem',
+            ],
+            'expected_adapter' => [
+               'class'   => \Zend\Cache\Storage\Adapter\Filesystem::class,
+               'options' => [
+                  'cache_dir' => GLPI_CACHE_DIR . '/_default',
+               ],
+               'plugins' => [
+                  \Zend\Cache\Storage\Plugin\Serializer::class
+               ],
+            ],
+         ],
+
+         // Case: filesystem adapter using custom namespace
+         [
+            'config'   => [
+                'adapter' => 'filesystem',
+                'options' => [
+                   'namespace' => 'my_cache',
+                ],
+            ],
+            'expected_adapter' => [
+               'class'   => \Zend\Cache\Storage\Adapter\Filesystem::class,
+               'options' => [
+                  'namespace' => 'my_cache',
+                  'cache_dir' => GLPI_CACHE_DIR . '/my_cache',
+               ],
+               'plugins' => [
+                  \Zend\Cache\Storage\Plugin\Serializer::class
+               ],
+            ],
+         ],
+
+         // Case: filesystem adapter using custom directory
+         [
+            'config'   => [
+                'adapter' => 'filesystem',
+                'options' => [
+                   'namespace' => 'my_cache',
+                   'cache_dir' => '/tmp/cache',
+                ],
+            ],
+            'expected_adapter' => [
+               'class'   => \Zend\Cache\Storage\Adapter\Filesystem::class,
+               'options' => [
+                  'namespace' => 'my_cache',
+                  'cache_dir' => '/tmp/cache',
+               ],
+               'plugins' => [
+                  \Zend\Cache\Storage\Plugin\Serializer::class
+               ],
+            ],
+         ],
+
+         // Case: redis adapter using default options
```

Comment:
This would require also a running redis instance. I'm not sure we have to check every possible provider. If the configuration is the one we expect, this should be enough; and you can just drop this block.

Reply:
We automatically add the `serializer` plugin if `redis` adapter is used, so it would be better if we test it.
I was thinking about creating a separate "cache" test suite that can be runned in an environment using a redis docker, a dba docker, ... But it will not be done now.

Reply:
Indeed, this can be added, but I do not think this is an emergency (I get no feedback about anyone using cache in glpi with something else than defaults)